dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -251,7 +282,105 @@ void RemoveFrameworkFromPath(PCWSTR frameworkPath)
+    {
+        return true;
+    }
+
+    // Use the AppExtension-style LifetimeManager
+    return false;
+}
+
+void CreateLifetimeManager(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion,
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager>& lifetimeManager,
+    wil::unique_event& endTheLifetimeManagerEvent,
+    wil::unique_cotaskmem_string& ddlmPackageFullName)
+{
+    if (IsLifetimeManagerViaEnumeration())
+    {
+        CreateLifetimeManagerViaEnumeration(majorMinorVersion, versionTag, minVersion, endTheLifetimeManagerEvent, ddlmPackageFullName);
+    }
+    else
+    {
+        CreateLifetimeManagerViaAppExtension(majorMinorVersion, versionTag, minVersion, lifetimeManager, ddlmPackageFullName);
+    }
+}
+
+void CreateLifetimeManagerViaAppExtension(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion,
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager>& lifetimeManager,
+    wil::unique_cotaskmem_string& ddlmPackageFullName)
+{
+    const auto appDynamicDependencyLifetimeManagerClsid{ FindDDLMViaAppExtension(majorMinorVersion, versionTag, minVersion) };
+
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager> dynamicDependencyLifetimeManager{
+        wil::CoCreateInstance<IDynamicDependencyLifetimeManager>(appDynamicDependencyLifetimeManagerClsid, CLSCTX_LOCAL_SERVER)
+    };
+
+    THROW_IF_FAILED(dynamicDependencyLifetimeManager->Initialize());
+
+    wil::unique_cotaskmem_string packageFullName;
+    THROW_IF_FAILED(dynamicDependencyLifetimeManager->GetPackageFullName(&packageFullName));
+
+    lifetimeManager = std::move(dynamicDependencyLifetimeManager);
+    ddlmPackageFullName = std::move(packageFullName);
+}
+
+void CreateLifetimeManagerViaEnumeration(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion,
+    wil::unique_event& endTheLifetimeManagerEvent,
+    wil::unique_cotaskmem_string& ddlmPackageFullName)
+{
+    std::wstring packageFamilyName;
+    std::wstring packageFullName;
+    FindDDLMViaEnumeration(majorMinorVersion, versionTag, minVersion, packageFamilyName, packageFullName);
+
+    // Create the named event used later to signal to the lifetime manager it's time to quit
+    // The named event has the syntax: "<processid>;<packagefullname>;<uniqueid>"
+    GUID uniqueId{};
+    THROW_IF_FAILED(CoCreateGuid(&uniqueId));
+    const auto c_uniqueIdAsString{ winrt::to_hstring(uniqueId) };
+    auto eventName{ wil::str_printf<wil::unique_cotaskmem_string>(L"%u;%s;%s", GetCurrentProcessId(), packageFullName.c_str(), c_uniqueIdAsString.c_str()) };
+    wil::unique_event event;
+    event.create(wil::EventOptions::ManualReset, eventName.get());
+
+    WCHAR lifetimeManagerApplicationUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH]{};
+    uint32_t lifetimeManagerApplicationUserModelIdLength{ static_cast<uint32_t>(ARRAYSIZE(lifetimeManagerApplicationUserModelId)) };
```

Comment:
Generally, `static_cast` of `ARRAYSIZE` shouldn't be necessary.

Reply:
Neat! For years that produced compiler warnings when I built x64 (ARRAYSIZE=size_t=uint64 but DWORD=uint32) but something must have changed as it now builds fine. Progress marches on :-)